commands/util/charts.js

```diff
@@ -1,28 +1,151 @@
     .addStringOption(option =>
       option.setName('icao')
-        .setDescription('The input to echo back')
+        .setDescription('The ICAO code of the airport')
+        .setRequired(true))
+    .addStringOption(option =>
+      option.setName('charttype')
+        .setDescription('The type of chart to retrieve')
         .setRequired(true)
-        .setMaxLength(4)),
-  async execute (interaction) {
-    const airport = interaction.options.getString('icao')
-    const response = await fetch(`https://metar.vatsim.net/metar.php?id=${airport}`)
-    const body = await response.text()
-
-    const chartsEmbed = new EmbedBuilder()
-      .setTitle('Airport')
-      .setDescription(`Information about ${airport.toUpperCase()}`)
-      .setColor('#37B6FF')
-      .addFields({ name: 'Charts (AirNav)', value: `https://www.airnav.com/airport/${airport}` },
-        { name: 'METAR', value: `${body}` || 'Not Available' })
-      .setFooter({ text: 'Made by TPC Dev Team', iconURL: 'https://static1.squarespace.com/static/614689d3918044012d2ac1b4/t/616ff36761fabc72642806e3/1634726781251/TPC_FullColor_TransparentBg_1280x1024_72dpi.png' })
-      .setTimestamp()
-
-    interaction.reply({ embeds: [chartsEmbed] })
-  }
-}
+        .addChoices(
+          { name: 'Airport Diagram', value: 'APD' },
+          { name: 'Departure Procedure', value: 'DP' },
+          { name: 'STARs', value: 'STAR' },
+          { name: 'Hot Spot', value: 'HOT' },
+          { name: 'Instrument Approach Procedures', value: 'IAP' },
+          { name: 'Land and Hold Short Procedures', value: 'LAH' },
+          { name: 'Minimums', value: 'MIN' },
+          { name: 'Obstacle Departure Procedures', value: 'ODP' },
+        )),
+  async execute(interaction) {
+    await interaction.deferReply();
+    try {
+      const icaoCode = interaction.options.getString('icao').toUpperCase();
+      const chartType = interaction.options.getString('charttype')?.toUpperCase();
+      console.log(`Executing command for ICAO code: ${icaoCode}`);
+            const diagrams = await airportDiagrams(icaoCode, chartType);
+
+            if (diagrams && diagrams.length > 0) {
+              const diagramsEmbed = new EmbedBuilder()
+                .setColor(0x0099FF)
+                .setTitle(`Airport Diagrams for ${icaoCode}`)
+                .setDescription('Here are the available diagrams:')
+                .addFields(diagrams.map(diagram => ({
+                  name: diagram.procedure.name,
+                  value: diagram.procedure.url,
+                  inline: true
+                })))
+                .setTimestamp();
+
+              await interaction.editReply({ embeds: [diagramsEmbed] });
+            } else {
+              await interaction.editReply(`${icaoCode.toUpperCase()} has no available charts online. [SkyVector may be able to provide more information.](https://skyvector.com/api/airportSearch?query=${icaoCode})`);
+            }
+          } catch (error) {
```

Comment:
When doing catches, please use the send to sentry function that way we know when things happen.